api/views.py

```diff
@@ -82,20 +84,112 @@ def post(self, request):
         return response.Response(status=status.HTTP_200_OK)
 
 
-class ProfileView(generics.RetrieveUpdateAPIView):
+class ProfileRetrieveUpdateView(generics.RetrieveUpdateAPIView):
     serializer_class = ProfileSerializer
 
     def get_object(self):
         return self.request.user.profile
 
     def get(self, request):
-        user = request.user
-        return response.Response(
-            data=ProfileSerializer(user.profile).data, status=status.HTTP_200_OK
+        profile = request.user.profile
+        profile_content = self.serializer_class(profile).data
+        userId = profile_content.pop("id")
+
+        # Add user interests to profile content.
+        interests = self.getUserInterests(profile)
+        profile_content["interests"] = interests
+
+        # Get privacy object related to user profile.
+        privacy = Privacy.objects.get(profile=profile)
+        privacy_settings = PrivacySerializer(privacy).data
+
+        data = {
+            "profile_content": profile_content,
+            "privacy_settings": privacy_settings,
+            "userId": userId,
+        }
+
+        return response.Response(data=data, status=status.HTTP_200_OK)
+
+    def put(self, request):
+        try:
+            # Updated user interests.
+            updatedInterests = request.data.pop("interests")
+            self.updateInterests(updatedInterests, request.user.profile)
+        except KeyError:
+            # User did not updates their interests.
+            pass
+
+        # Update profile content.
+        return self.update(request=request)
```

Comment:
This is related to the previous comment and to what we discussed the other day. This would be simpler by taking advantage of a many-to-many field on the profile model.